include/cfdcore/cfdcore_json_mapping_base.h

```diff
@@ -135,11 +135,20 @@ inline void ConvertFromUniValue(
 template <typename T>
 inline void ConvertFromUniValue(
     T& value, const UniValue& json_value) {  // NOLINT
-  if (json_value.isNum()) {
+  UniValue json_value_copy = json_value;
+  if (json_value_copy.isStr()) {
+    std::string str = json_value.get_str();
+    bool is_digits_only = std::all_of(str.begin(), str.end(), ::isdigit);
```

Comment:
If use the `isdigit` function, it will reject negative values. Negative values​should move the check start position.
```suggestion
    auto begin_pos = str.begin();
    if (*begin_pos == '-') ++begin_pos;
    bool is_digits_only = std::all_of(begin_pos, str.end(), ::isdigit);
```

Reply:
Good point thanks, updated.